skfda/clustering.py

```diff
@@ -0,0 +1,1180 @@
+        used in the :func:`fit <fda.clustering.ClusteringData.fit>` function.
+
+        Args:
+            fdatagrid (FDataGrid object): Object whose samples
+                are classified into different groups.
+            n_clusters (int): Number of groups into which the samples are
+                classified.
+            init (ndarray): Contains the initial centers of the different
+                clusters the algorithm starts with or None.
+
+        Returns:
+            fdatagrid (FDataGrid object): Object whose samples
+                are classified into different groups.
+            n_clusters (int): Number of groups into which the samples are
+                classified.
+            init (ndarray): Contains the initial centers of the different
+                clusters the algorithm starts with. If None, default value,
+                the centers are generated randomly.
+        """
+
+        if fdatagrid.ndim_domain > 1:
+            raise NotImplementedError(
+                "Only support 1 dimension on the domain.")
+
+        if fdatagrid.nsamples < 2:
+            raise ValueError(
+                "The number of observations must be greater than 1.")
+
+        if n_clusters < 2:
+            raise ValueError(
+                "The number of clusters must be greater than 1.")
+
+        if init is not None and init.shape != (
+                fdatagrid.ndim_image, n_clusters, fdatagrid.ncol):
+            raise ValueError(
+                "The init ndarray should be of shape (ndim_image, n_clusters, n_features) "
+                "and gives the initial centers.")
+        elif init is None:
+            init = np.array([None] * fdatagrid.ndim_image)
+
+        return fdatagrid, n_clusters, init
+
+    def _random_initialization_centers(self, data_matrix, fdatagrid,
+                                       n_clusters):
+        """Random initialization of the centroids used in the
+        :func:`fit function <fda.clustering.ClusteringData.fit>` if *init*
+        is None.
+
+        Args:
+            data_matrix (ndarray): matrix with the data only of the
+                dimension of the image of the fdatagrid the algorithm is
+                classifying.
+            fdatagrid (FDataGrid object): Object whose samples are
+                classified into different groups.
+            n_clusters (int): Number of groups into which the samples are
+                classified.
+
+        Returns:
+            centers (ndarray): initial centers
+        """
+        comparison = True
+        random_state_aux = self.random_state
+        while comparison:
+            random.seed(a=random_state_aux)
+            indices = random.sample(range(fdatagrid.nsamples), n_clusters)
+            centers = data_matrix[indices]
+            comparison = np.asarray([(centers[i] == centers).all()
+                                     for i in range(n_clusters)]).sum()
+            random_state_aux += 1
```

Comment:
Can this be done for RandomState objects?

Reply:
If the new code is the proposed in the below question, the parameter of the class would only accept random_seed, which in that case, it is an Int.